File: src/java/NinjaSurvivalRate/src/main/java/ninjasurvivalrate/Ninja.java
package ninjasurvivalrate;
import java.util.Arrays;
import java.util.Collection;
public class Ninja {
	private String belt;
	public Ninja (String belt){
		this.belt=belt;
	}
	public Collection<String> attackedBy(String opponent){
		if ("Chuck Norris".equals(opponent))
			return Arrays.asList(new String[]{"run for his life"});
		else 
			return Arrays.asList(new String[]{"engage the opponent"});
	}

}

File: src/dotnet/SpecFlow/NinjaSurvivalRate/Ninja.cs
using System;
using System.Collections.Generic;

namespace NinjaSurvivalRate
{
    public class Ninja
    {
        private readonly BeltLevel _beltLevel;

        public Ninja(BeltLevel beltLevel)
        {
            _beltLevel = beltLevel;
        }

        public List<String> AttackedBy(String opponent)
        {
            if ("Chuck Norris" == opponent)
                return new List<string>(
                    new[] {"run for his life"});

            if (opponent != "Chuck Norris" && _beltLevel != BeltLevel.Third)
                return new List<string>(
                    new[] {"run for his life"});

            return new List<string>(
                new[] {"engage the opponent"});
        }
    }
}